src/cache/tlru-cache.spec.ts

```diff
@@ -0,0 +1,159 @@
+    jest.useFakeTimers();
+
+    cache.set('a', 'avocado');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs);
+    cache.set('b', 'banana');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs);
+
+    const cacheValues = [];
+
+    for (const value of cache.values()) {
+      cacheValues.push(value);
+    }
+
+    expect(cacheValues.length).toBe(0);
+  });
+
+  it('should overwrite existing cache entry', () => {
+    jest.useFakeTimers();
+
+    cache.set('a', 'apple');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs - 1);
+    cache.set('a', 'avocado');
+
+    // spin the clock by 5sec. After that time cache entry should be still valid.
+    jest.advanceTimersByTime(expectedCacheTimeoutMs / 2);
+
+    // setting assertion in a weird way because calling cache.get()
+    // will reset eviction timer which will mess up next assertion
+    let avocadoInCache = false;
+    cache.forEach((value, key) => {
+      if (key === 'a' && value === 'avocado') {
+        avocadoInCache = true;
+      }
+    });
+    expect(avocadoInCache).toBe(true);
+
+    // after another spin of 5 sec, cache entry should evict itself
+    jest.advanceTimersByTime(expectedCacheTimeoutMs / 2);
+    expect(cache.get('a')).toBeUndefined();
+  });
+
+  it('should check if a key exists', () => {
+    cache.set('a', 'apple');
+    expect(cache.has('a')).toBeTruthy();
+    expect(cache.has('b')).toBeFalsy();
+  });
+
+  it('should handle the cache capacity of zero', () => {
+    const zeroCache = new TLRUCache(0, expectedCacheTimeoutMs);
+    zeroCache.set('a', 'apple');
+    expect(zeroCache.get('a')).toBeFalsy();
+  });
+
+  it('should handle the cache capacity of one', () => {
+    jest.useFakeTimers();
+    const oneCache = new TLRUCache(1, expectedCacheTimeoutMs);
+    oneCache.set('a', 'apple');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs);
+    expect(oneCache.get('a')).toBeUndefined();
+
+    oneCache.set('a', 'avocado');
+    expect(oneCache.get('a')).toBe('avocado');
+    oneCache.set('b', 'banana');
+    expect(oneCache.get('a')).toBeFalsy();
+    expect(oneCache.get('b')).toBe('banana');
+  });
+
+  it('should evict oldest entry when capacity limit is reached', () => {
+    cache.set('a', 'apple');
+    cache.set('b', 'banana');
+    cache.set('c', 'cherry');
+
+    expect(cache.get('a')).toBeUndefined();
+    expect(cache.has('b')).toBeTruthy();
+    expect(cache.has('c')).toBeTruthy();
+  });
+
+  /**
+    This test case might be an overkill but in case Map() changes,
+    or we want to ditch it completely this will remind us that insertion
+    order is crucial for this cache to work properly
+  **/
+  it('should preserve insertion order when inserting on capacity limit', () => {
+    cache.set('a', 'apple');
+    cache.set('b', 'banana');
+    cache.set('c', 'cherry');
+
+    const keys = Array.from(cache.keys());
+    expect(keys[0]).toBe('b');
+    expect(keys[1]).toBe('c');
+  });
```

Comment:
I think we need another test that does:
```
cache.set('a', 'apple');
cache.set('b', 'banana');
cache.get('a');
cache.set('c', 'cherry');
```
Then makes sure its `c` and `a` (e.g., a got re-inserted at the end)